### PR TITLE
www-apache/modsecurity-crs: add 3.3.4

### PR DESCRIPTION
**Pull request by hydrapolic**
Add apache2/nginx USE flags so nginx users don't have to pull in apache2. This needs https://github.com/gentoo/gentoo/pull/27884 and https://github.com/gentoo/gentoo/pull/27885 where we added the new versions.

```sh
--- /usr/portage/www-apache/modsecurity-crs/modsecurity-crs-3.3.2.ebuild        2021-12-22 01:10:03.000000000 +0000
+++ modsecurity-crs-3.3.4.ebuild        2022-10-22 04:00:19.688251423 +0000
@@ -1,4 +1,4 @@
-# Copyright 1999-2021 Gentoo Authors
+# Copyright 1999-2022 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
 EAPI="8"
@@ -6,28 +6,37 @@
 DESCRIPTION="OWASP ModSecurity Core Rule Set"
 HOMEPAGE="https://coreruleset.org/"
 SRC_URI="https://github.com/coreruleset/coreruleset/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+S="${WORKDIR}/coreruleset-${PV}"
 
 LICENSE="Apache-2.0"
 SLOT="0"
-KEYWORDS="amd64 x86"
+KEYWORDS="~amd64 ~x86"
+IUSE="+apache2 nginx"
 
-RDEPEND=">=www-apache/mod_security-2.9.1"
+RDEPEND="
+       apache2? ( >=www-apache/mod_security-2.9.6 )
+       nginx? ( >=dev-libs/modsecurity-3.0.8 )
+"
 
-S="${WORKDIR}/coreruleset-${PV}"
+DOCS=( CHANGES CONTRIBUTORS.md crs-setup.conf.example KNOWN_BUGS README.md )
 
 src_install() {
        insinto "/usr/share/${PN}"
        doins -r rules
 
-       dodoc CHANGES CONTRIBUTORS.md KNOWN_BUGS README.md crs-setup.conf.example
+       einstalldocs
 
-       # I don't think it's worth pulling in apache-module.eclass just for
-       # this path...
-       insinto /etc/apache2/modules.d
-       doins "${FILESDIR}/80_mod_security-crs.conf"
+       if use apache2; then
+               # I don't think it's worth pulling in apache-module.eclass just for
+               # this path...
+               insinto /etc/apache2/modules.d
+               doins "${FILESDIR}/80_mod_security-crs.conf"
+       fi
 }
 
 pkg_postinst() {
-       einfo "The CRS configuration file has been installed to:"
-       einfo "  ${ROOT}/etc/apache2/modules.d/80_mod_security-crs.conf"
+       if use apache2; then
+               einfo "The CRS configuration file has been installed to:"
+               einfo "  ${ROOT}/etc/apache2/modules.d/80_mod_security-crs.conf"
+       fi
 }
```

### CODE REVIEW

**Comment by hydrapolic** [comment]
> ## Pull request CI report
> 
> _Report generated at_: 2022-10-22 04:23 UTC _Newest commit scanned_: [eaae505](https://github.com/gentoo/gentoo/commit/eaae5054fc6065dc7959d4b64001ad432c0f23e8) _Status_: x **broken**
> 
> New issues caused by PR: https://qa-reports.gentoo.org/output/gentoo-ci/c67bd738dd/output.html#www-apache/modsecurity-crs
> 
> There are existing issues already. Please look into the report to make sure none of them affect the packages in question: https://qa-reports.gentoo.org/output/gentoo-ci/c67bd738dd/output.html

This is only missing versions from other PRs as mentioned in the summary.

**Comment by hydrapolic** [comment]
Using in production.

**Review by juippis — COMMENTED** [review]
In future it'd be great if you bundled these related commits / PRs to a single one! Please ping me / us when the deps have been merged since I've now lost the tracking through mail client :P

**Comment by hydrapolic** [comment]
> In future it'd be great if you bundled these related commits / PRs to a single one! Please ping me / us when the deps have been merged since I've now lost the tracking through mail client :P

I didn't have it prepared all once once, so I started with one piece, opened a PR and them moved to the other one. Didn't want to update 1 PR while still working on it while the others could be merged in the meanwhile.

**Comment by hydrapolic** [comment]
> In future it'd be great if you bundled these related commits / PRs to a single one! Please ping me / us when the deps have been merged since I've now lost the tracking through mail client :P

The missing deps are now in portage, this can be merged @juippis.

**Review by juippis — APPROVED** [review]
Looks good, just one question though: should the `+nginx` use flag do something here? Change installation paths, add configs...?

Otherwise I'm going to migrate it into an *optfeature* as it should.

**Comment by hydrapolic** [comment]
> Looks good, just one question though: should the `+nginx` use flag do something here? Change installation paths, add configs...?
> 
> Otherwise I'm going to migrate it into an _optfeature_ as it should.

It's there to not bring in apache when you only have nginx installed. This way you can have nginx+modsecurity without apache.

File: www-apache/modsecurity-crs/modsecurity-crs-3.3.4.ebuild [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+S="${WORKDIR}/coreruleset-${PV}"
+
+LICENSE="Apache-2.0"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="+apache2 nginx"
+
+RDEPEND="
+	apache2? ( >=www-apache/mod_security-2.9.6 )
+	nginx? ( >=dev-libs/modsecurity-3.0.8 )
```
Comment (juippis) [inline www-apache/modsecurity-crs/modsecurity-crs-3.3.4.ebuild:18]:
```suggestion
	apache2? ( >=www-apache/mod_security-2.9.6 )
	!apache? ( >=dev-libs/modsecurity-3.0.8 )
```
in that case would this make sense?

Comment (hydrapolic) [reply www-apache/modsecurity-crs/modsecurity-crs-3.3.4.ebuild:18]: I would find it more suprising as seeing apache2 / nginx use flags.

**Review by juippis — APPROVED** [review]
I'll switch to optfeature, thanks!

**Comment by hydrapolic** [comment]
Thanks @juippis

**Pull request closed without merging**